data/test/scenarios/test_terrain_mask.cfg

```diff
@@ -13,139 +13,281 @@ raw#endarg
-"Hh, Hh
+    "Hh, Hh
 Hh, Hh"#enddef
 
-# Alpha and layering works
+#####
+# API(s) being tested: [terrain_mask]
+##
+# Actions:
+# Apply a terrain mask of _f to a hex.
+##
+# Expected end state:
+# The terrain doesn't change.
+#####
 {TERRAIN_MASK_TEST simple_nop "_f" (POS=9,1) (PRE_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Kov}) (POST_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Kov})}
+#####
+# API(s) being tested: [terrain_mask]
+##
+# Actions:
+# Apply a terrain mask of Gd to a hex.
+##
+# Expected end state:
+# The terrain on that location changes to Gd.
+#####
 {TERRAIN_MASK_TEST simple_set "Gd" (POS=9,1) (PRE_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Kov}) (POST_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gd})}
+#####
+# API(s) being tested: [terrain_mask]
+##
+# Actions:
+# Apply a terrain mask of Gd^Vo to a hex.
+# Use a rule for only the base terrain.
+##
+# Expected end state:
+# The base terrain of the hex is changed, but the overlay is not.
+#####
 {TERRAIN_MASK_TEST simple_underlay "Gd^Vo" (POS=9,1) (PRE_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Kov}) (POST_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gd^Kov}) (RULES={TERRAIN_RULE_LAYER base})}
+#####
+# API(s) being tested: [terrain_mask]
+##
+# Actions:
+# Apply a terrain mask of Gd^Vo to a hex that already has a base terrain of Gd.
+# Use a rule for only the base terrain.
+##
+# Expected end state:
+# The terrain of the hex is not changed.
+#####
 {TERRAIN_MASK_TEST simple_overlay "Gd^Vo" (POS=9,1) (PRE_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Kov}) (POST_MASK={ASSERT_LOCATION_TERRAIN 9 1 Gg^Vo}) (RULES={TERRAIN_RULE_LAYER overlay})}
 
 # Location IDs from the mask are added to the map, unless this is explicitly disabled
+#####
+# API(s) being tested: [terrain_mask]
+##
+# Actions:
+# Add a location id to a hex.
+##
+# Expected end state:
+# The hex has the location id associated to it.
+#####
 {TERRAIN_MASK_TEST simple_name "foo Gg" (POS=9,1) (POST_MASK={ASSERT_LOCATION_NAME 9 1 foo}) (PRE_MASK=
-    {ASSERT (
-        [not]
-            [have_location]
-                x,y=9,1
-                location_id=foo
-            [/have_location]
-        [/not]
-    )}
+{ASSERT (
+    [not]
+        [have_location]
+            x,y=9,1
+            location_id=foo
+        [/have_location]
+    [/not]
+)}
 )}
+#####
+# API(s) being tested: [terrain_mask]ignore_special_locations=
+##
+# Expected end state:
+# The hex is not found by location id since locations are not set by the terrain mask.
+#####
 {TERRAIN_MASK_TEST simple_name_fail "foo Gg" (POS=9,1) (POST_MASK=
-    [+terrain_mask]
-        ignore_special_locations=yes
-    [/terrain_mask]
-    {ASSERT (
-        [not]
-            [have_location]
-                x,y=9,1
-                location_id=foo
-            [/have_location]
-        [/not]
-    )}
+[+terrain_mask]
+    ignore_special_locations=yes
+[/terrain_mask]
+{ASSERT (
+    [not]
+        [have_location]
+            x,y=9,1
+            location_id=foo
+        [/have_location]
+    [/not]
+)}
```

Comment:
This file (in other parts too) is looking worse for having wmlindent run on it.

Reply:
Alright, I'm going to merge then. I went through and reverted a couple changed files since wmlindent definitely wasn't interpreting some of the lua parts correctly, but otherwise this is how it's handled in the rest of the codebase.